store/redis_supplier_roles.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package store
+
+import (
+	"context"
+	"fmt"
+
+	l4g "github.com/alecthomas/log4go"
+
+	"github.com/mattermost/mattermost-server/model"
+)
+
+func (s *RedisSupplier) RoleSave(ctx context.Context, role *model.Role, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
+	if err := s.client.Del(buildRedisKeyForRoleName(role.Name)).Err(); err != nil {
+		l4g.Error("Redis failed to remove key roles:" + role.Name + " Error: " + err.Error())
```

Comment:
You are using roles:" + role.Name to build the RedisKeyForRoleName value, looks like is better use the redis key in a previously declared variable and use in the error if needed.